src/generic/grideditors.cpp

```diff
@@ -1266,14 +1245,26 @@ void wxGridCellBoolEditor::SetSize(const wxRect& r)
 
     // position it in the centre of the rectangle (TODO: support alignment?)
 
+    wxRect cellRect = r;
+
 #if defined(__WXGTK__) || defined (__WXMOTIF__)
+#if defined(__WXGTK3__)
+    cellRect.x += 6;
+    cellRect.width -= 12;
+#else
     // the checkbox without label still has some space to the right in wxGTK,
     // so shift it to the right
+    // TODO: check offsets
+    cellRect.x += 2;
+    cellRect.width -= 2;
     size.x -= 8;
+#endif
 #elif defined(__WXMSW__)
-    // here too, but in other way
-    size.x += 1;
-    size.y -= 2;
+    cellRect.x += 2;
+    cellRect.width -= 2;
```

Comment:
This would also be more clear to write as a combination of `Deflate(1, 0)` and `Offset(1, 0)`.

But mostly it would be great to get rid of it, instead of tweaking it.